### DISCUSSION TITLE
imprt big database

### DISCUSSION THREAD

**Discussion by siamakabi** [pocketbase/pocketbase#3227] [Q&A]
Hello
 How to import a big database (4TB) ?

**Comment by ganigeorgiev** [comment]
I don't think there is a universal answer to this.

There is no builtin mechanism for importing external data and this will highly depend on your data structure and the exported format.

In general large datasets can be imported by:

- creating a SQLite scripts and executing them against pb_data/data.db (this will also require to create the collections for your data either from the Admin UI or programmatically via [JS](https://pocketbase.io/docs/js-collections/) or [GO](https://pocketbase.io/docs/go-collections/); _auto table discovery will be supported in the future_)

- or import them programmatically via custom JS or Go command. For a JS command example check https://github.com/pocketbase/pocketbase/discussions/3129#discussioncomment-6720708.

**Reply by siamakabi** [reply]
thanks for your answer
If it is possible to import the Json file through the admin panel, the problem of many people will be solved

**Comment by oschettler** [comment]
Not an answer, but the same question: I am about to migrate a MySQL/PHP app to Pocketbase. I can convert the mysql database to sqlite. 

From there, is manually creating the necessary collections, using the admin UI, still the way to go?

**Reply by ganigeorgiev** [reply]
Yes. PocketBase has its own specific way to organize the tables data and direct mapping may not always work (for example we don't support `NULL` for the majority of the fields). I recommend to explore the [available collection fields](https://pocketbase.io/docs/collections/#fields) and then decide whether you want to invest time to migrate or not.

After that, a safer way to import existing data is to insert it as Record model since with v0.23+ this will also run the necessary validations and data integrity checks (you can find an example `.json` file import command in https://github.com/pocketbase/pocketbase/discussions/6040#discussioncomment-11490098).

_As a side-note: keep in mind that PocketBase is not stable yet and breaking changes should be expected, so if you already have a working MySQL/PHP app then I'd advice against migrating to PocketBase if you don't have a good reason for that because most likely you'll be disappointed. See also the note in the [FAQ](https://pocketbase.io/faq/)._

**Reply by ganigeorgiev** [reply]
_Side-note 2: If you still want to migrate, I forgot to mention that another alternative could be to create a small Go application that will connect to both MySQL and PocketBase (this also allows migrating uploaded files). It is a little out-dated but you could find one such example in https://github.com/presentator/v2tov3migrate (Presenator is my other open source project that similarly as in your case v2 was using MySQL/PHP and v3 has been migrated to PocketBase)._

**Reply by oschettler** [reply]
> insert it as Record model

Thank your for the kind advice. I will follow this route then. 

I am attracted to sqlite as opposed to mysql because of the lower resource requirements. Also, I have noticed during the last 5+ years, that Laravel (the PHP framework I used), together with the required upgrades to the PHP language, is a very moving target as well.
